lib/matplotlib/legend.py

```diff
@@ -558,8 +558,7 @@ def val_or_rc(val, rc_name):
                                        colors.to_rgba_array(labelcolor))):
                 text.set_color(color)
         else:
-            raise ValueError("Invalid argument for labelcolor : %s" %
-                             str(labelcolor))
+            raise ValueError(f"Invalid argument for labelcolor: {labelcolor}")
```

Comment:
```suggestion
            raise ValueError(f"Invalid argument for labelcolor: {labelcolor!r}")
```
See #21959 (and #22712 )

Reply:
sure